packages/node-firebird-driver/src/lib/impl/attachment.ts

```diff
@@ -100,11 +100,33 @@ export abstract class AbstractAttachment implements Attachment {
 				prepareOptions?: PrepareOptions,
 				executeOptions?: ExecuteOptions
 			}): Promise<Array<any>> {
+		return (await this.executeReturningAs<Array<any>>(transaction, sqlStmt, parameters, {
+			...options, json: false
+		})).row;
+	}
+
+	async executeReturningAs<T>(transaction: AbstractTransaction, sqlStmt: string, parameters?: Array<any>,
+						   options?: {
+							   prepareOptions?: PrepareOptions,
+							   executeOptions?: ExecuteOptions,
+							   json?: boolean
+						   }): Promise<{ row: T, cols: Array<string> }> {
 		this.check();
 
-		const statement = await this.prepare(transaction, sqlStmt, options && options.prepareOptions);
+		const statement = await this.prepare(transaction, sqlStmt, options?.prepareOptions);
+
 		try {
-			return await statement.executeReturning(transaction, parameters, options && options.executeOptions);
+			const cols = (await statement?.columnLabels) || [];
+			const rowColVal = await statement.executeReturning(transaction, parameters, options?.executeOptions);
+			let rowObj: any;
+			if (options?.json) {
+				rowObj = {} as any;
+				rowColVal.forEach((v, idx) => rowObj[cols[idx]] = v);
```

Comment:
I'm not sure columnsLabels should be used directly. That means we'll in most cases create objects with upper-case and underline separators. Maybe we should transform, for example, `'COLUMN_NAME'` to `'columnName'`?

Reply:
I don't agree. Column labels are actually column ids. If we modify them to match code styles we lose their integrity.